controllers/vrg_volrep.go

```diff
@@ -1940,11 +1940,7 @@ func (v *VRGInstance) addPVRestoreAnnotation(pv *corev1.PersistentVolume) {
 //
 //nolint:funlen
 func (v *VRGInstance) aggregateVolRepDataReadyCondition() *metav1.Condition {
-	if len(v.volRepPVCs) == 0 {
-		return nil
-	}
-
-	vrgReady := len(v.instance.Status.ProtectedPVCs) != 0
+	vrgReady := true
```

Comment:
I think this is dangerous.  If the PVC hasn't gone to a bound state, it will not be in the ProtectedPVCs. With this change, we will falsely mark the VRG as `DataProtected`.

Reply:
Ok, so that is the reason for checking the length of ProtectedPVCs.

The bug that I am trying to fix is the one where there are no PVCs to protect at all. Would looking at the v.volRepPVCs and v.volSyncPVCs and setting vrgReady to true if they are both of length 0 acceptable?

Reply:
> Would looking at the v.volRepPVCs and v.volSyncPVCs and setting vrgReady to true if they are both of length 0 acceptable?
That will do it. However, I don't believe that's the right approach. Why should the `DataReady` be true when there are no PVCs? I think it should remain false to indicate that data is partially or fully not ready. Let's think of this scenario; the app was deployed correctly, but the PVCs didn't, for whatever reason (i.e. The app PVCs are not available in the git repo pointed to by the channel). In that case, the size of `v.volRepPVCs` will be 0. So the app is not running (waiting for the PVCs) but the VRG is reporting that all is good.

Reply:
Maybe it is better to fix the unit test than try to mess with the `DataReady` condition.  That area is very fragile.  We need to refactor it sooner rather than later.

Reply:
The test case is something like this:

App has 3 PVCs of which they only want DR protection for one. Lets say that the other two PVs are only used for storing temp data. They use the label selector to indicate which of the PVCs must have DR protection. Also, the DR protection PVC isn't created till a certain app user action(say creating a new project in the app).

In this case, we should advertise the App as DR ready under the following cases:
1. No PVCs exist
2. Only the 2 temp PVCs exist
3. All 3 PVCs exist and the DR required PVC is bound and protected

In the current code, Ramen sets DataReady as false for case 1 and case 2. Here is the bug I am trying to fix https://bugzilla.redhat.com/show_bug.cgi?id=2128142

Reply:
They attempted relocation. So even if you advertise DataReay as true, the relocation will fail because of `DataProtected`

Reply:
In your test case, we should not advertise `DataReady` as true given that the app has at least 1 PVC that they want to protect.  Why do they need `DataReady` for?